reconcile/gitlab_owners.py

```diff
@@ -45,6 +45,17 @@ def __init__(self, gitlab_client, merge_request, owners, dry_run, persistent_lgt
             top_commit = next(commits)
             self.top_commit_created_at = dateparser.parse(top_commit.created_at)
 
+    def expand_groups(self, owners: list[str] | None) -> None:
+        if owners:
+            members = set()
+            for name in owners:
+                if group := self.gitlab.get_group_if_exists(name):
+                    members.update(self.gitlab.get_group_members(group))
+            for member in members:
+                name = member.username
+                if name not in owners:
+                    owners.append(name)
```

Comment:
since only `username` is required, and `get_group_members` returns a complex object `GroupMember`, better to just grab `username` to set like

```python
    members.update(group_member.username for group_member in self.gitlab.get_group_members(group))
```

`name not in owners` inside `members` loop check is O(N^2), can return set union directly


```python
return list(members.union(owners))
```

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4906/commits/18d83cef230bda319d0bea3ca5d208823050c438